pkg/plugins/policies/meshtrace/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,113 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"golang.org/x/exp/slices"
+
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshTraceResource) validate() error {
+	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.GetTargetRef()))
+	verr.AddErrorAt(path.Field("default"), validateDefault(r.Spec.GetDefault()))
+	return verr.OrNil()
+}
+func validateTop(targetRef *common_proto.TargetRef) validators.ValidationError {
+	targetRefErr := matcher_validators.ValidateTargetRef(targetRef, &matcher_validators.ValidateTargetRefOpts{
+		SupportedKinds: []common_proto.TargetRef_Kind{
+			common_proto.TargetRef_Mesh,
+			common_proto.TargetRef_MeshSubset,
+			common_proto.TargetRef_MeshService,
+			common_proto.TargetRef_MeshServiceSubset,
+			common_proto.TargetRef_MeshGatewayRoute,
+		},
+	})
+	return targetRefErr
+}
+
+func validateDefault(conf *MeshTrace_Conf) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if conf == nil {
+		verr.AddViolation("", validators.MustBeDefined)
+		return verr
+	}
+
+	if len(conf.GetBackends()) != 1 {
+		verr.AddViolation("backends", "must have exactly one backend defined")
```

Comment:
I think 0 backends is also allowed

Reply:
Does it make sense? If we're not sending it anywhere then it's pretty useless :/

Reply:
When we will have a proper `Merge` and the list will replace another list (instead of append) then an empty list of backends is a way to turn off tracing for the set of proxies.

For example, you have a global policy:
```
type: MeshTrace
name: global
spec:
  targetRef:
    kind: Mesh
  default:
    backend: 
      - datadog: {}
 ```
 and using `exclude-backend` policy you can turn off tracing only for `backend` service:
 ```
 type: MeshTrace
name: exclude-backend
spec:
  targetRef:
    kind: MeshService
    name: backend
  default:
    backend: []
 ```